cirq-core/cirq/transformers/diagonal_optimization.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transformer pass that removes diagonal gates before measurements."""
+
+from __future__ import annotations
+
+import cirq
+from cirq import ops, protocols
+from cirq.transformers import transformer_api
+from cirq.transformers.eject_z import eject_z
+
+
+def _is_diagonal(op: cirq.Operation) -> bool:
```

Comment:
Perhaps _is_z_or_cz_pow_gate is a better name because it conveys better what it checks.